save-backend/src/main/kotlin/org/cqfn/save/backend/controllers/CloneRepositoryController.kt

```diff
@@ -92,20 +108,21 @@ class CloneRepositoryController(
             log.info("Sending request to preprocessor to start save file for project id=${project.id}")
             val bodyBuilder = MultipartBodyBuilder()
             bodyBuilder.part("executionRequestForStandardSuites", executionRequestForStandardSuites)
-            return Mono.zip(propertyFile, binaryFile).map {
-                bodyBuilder.part("property", it.t1)
-                bodyBuilder.part("binFile", it.t2)
-            }.flatMap {
-                preprocessorWebClient
-                    .post()
-                    .uri("/uploadBin")
-                    .contentType(MediaType.MULTIPART_FORM_DATA)
-                    .body(BodyInserters.fromMultipartData(bodyBuilder.build()))
-                    .retrieve()
-                    .toEntity(String::class.java)
-                    .toMono()
+            return files.map {
```

Comment:
This logic is almost the same as in `submitExecutionRequest`, didn't you think about to move it in some function?

Reply:
Extracted

Reply:
And what about `preprocessorWebClient.post()...` chain? They differs only in `uri`. Or I missed something?

Reply:
You are right, I did some more refactoring

Reply:
Now it's looks even more better, than I expect.

Just one small question, why do you decide to create `ExecutionRequestBase` class? Previously we use ExecutionRequest and had no problems

Reply:
Because one version of the method uses `ExecutionRequest` and the other - `ExecutionRequestForStandardSuites`. So instead of passing a number of different arguments, I decided to introduce a base class, which could contain all those values.